File: lib/ftw_realtime/ai/fair_price_entry.ex
defmodule FtwRealtime.AI.FairPriceEntry do
  use Ecto.Schema
  import Ecto.Changeset

  @primary_key {:id, :binary_id, autogenerate: true}

  schema "fair_prices" do
    field(:category, :string)
    field(:zip_prefix, :string)
    field(:size, :string)
    field(:low, :integer)
    field(:high, :integer)
    field(:materials_pct, :float)
    field(:labor_pct, :float)
    field(:confidence, :string)
    field(:raw_response, :string)

    timestamps(type: :utc_datetime)
  end

  def changeset(entry, attrs) do
    entry
    |> cast(attrs, [
      :category,
      :zip_prefix,
      :size,
      :low,
      :high,
      :materials_pct,
      :labor_pct,
      :confidence,
      :raw_response
    ])
    |> validate_required([:category, :zip_prefix, :size, :low, :high])
    |> unique_constraint([:category, :zip_prefix, :size])
  end
end

File: src/main/kotlin/com/strata/ftw/domain/entity/FairPriceEntry.kt
package com.strata.ftw.domain.entity

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import java.util.UUID

@Entity
@Table(
    name = "fair_prices",
    uniqueConstraints = [UniqueConstraint(columnNames = ["category", "zip_prefix", "size"])]
)
class FairPriceEntry(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "uuid")
    var id: UUID? = null,

    @Column(nullable = false)
    var category: String = "",

    @Column(name = "zip_prefix", nullable = false)
    var zipPrefix: String = "",

    @Column(nullable = false)
    var size: String = "",

    @Column(nullable = false)
    var low: Int = 0,

    @Column(nullable = false)
    var high: Int = 0,

    @Column(name = "materials_pct")
    var materialsPct: Double? = null,

    @Column(name = "labor_pct")
    var laborPct: Double? = null,

    var confidence: String? = null,

    @Column(name = "raw_response")
    var rawResponse: String? = null,

    @CreationTimestamp
    @Column(name = "inserted_at", updatable = false)
    var insertedAt: Instant? = null,

    @UpdateTimestamp
    @Column(name = "updated_at")
    var updatedAt: Instant? = null
)
